pkg/cmd/cli/cmd/volume.go

```diff
@@ -0,0 +1,602 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kapierrors "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/meta"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	volumeLong = `Update volumes on a pod
+
+This command can add, update, list or remove volumes from containers in pods or
+any object that has a pod template (replication controllers or deployment configurations).
+You can specify a single object or multiple, and alter volumes on all containers or
+just those that match a wildcard.`
+
+	volumeExample = `  // Add new volume of type 'emptyDir' for deployment config 'registry' and mount under /opt inside the containers
+  // The volume name is auto generated
+  $ %[1]s volume dc/registry --add --mount-path=/opt
+
+  // Add new volume 'v1' with secret 'magic' for pod 'p1'
+  $ %[1]s volume pod/p1 --add --name=v1 -m /etc --type=secret --secret-name=magic
+
+  // Add new volume to pod 'p1' based on gitRepo (or other volume sources not supported by --type)
+  $ %[1]s volume pod/p1 --add -m /repo --source=<json-string>
+
+  // Add emptyDir volume 'v1' to a pod definition on disk and update the pod on the server
+  $ %[1]s volume -f pod.json --add --name=v1
+
+  // Create a new persistent volume and overwrite existing volume 'v1' for replication controller 'r1'
+  $ %[1]s volume rc/r1 --add --name=v1 -t persistentVolumeClaim --claim-name=pvc1 --overwrite
+
+  // Change pod 'p1' mount point to /data for volume v1
+  $ %[1]s volume pod p1 --add --name=v1 -m /data --overwrite
+
+  // Remove all volumes for pod 'p1'
+  $ %[1]s volume pod/p1 --remove --confirm
```

Comment:
I know there was some discussions around `--confirm` already, but: what happens if you don't specify `--confirm` here?